crates/isograph_lang_parser/src/peekable_lexer.rs

```diff
@@ -42,6 +42,9 @@ impl<'source> PeekableLexer<'source> {
     /// Get the next token (and advance)
     pub fn parse_token(&mut self) -> WithSpan<IsographLangTokenKind> {
         // Skip over (and record) any invalid tokens until either a valid token or an EOF is encountered
+        //
+        // Remove this allow after logic changed.
+        #[allow(clippy::never_loop)]
```

Comment:
Do you understand why clippy is complaining here?

Reply:
Since this block returns directly which leads to `never_loop`. For example:

```rust
pub fn parse_token(&mut self) -> WithSpan<IsographLangTokenKind> {
    loop {
        let kind = self
            .lexer
            .next()
            .unwrap_or(IsographLangTokenKind::EndOfFile);
        match kind {
            IsographLangTokenKind::Error => {
                panic!() // panic happened here.
            }
            _ => {
                // return happened here.
                return abc;
            }
        }
    }
}
```

I've left `allow` here, assuming you'll add more logic that requires the loop.

Reply:
Ah I see, makes sense. Thanks for highlighting this, yes, I believe we want the loop there. The Relay code does stuff in that branch, but it's not a high priority for me to recover from errors at the moment: https://github.com/facebook/relay/blob/a1ce4999fd68e7bcc66d542be5d91daa1a602356/compiler/crates/graphql-syntax/src/parser.rs#L2901-L2923